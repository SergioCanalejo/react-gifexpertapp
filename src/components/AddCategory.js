import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const handleInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const newCategory = inputValue.trim();
    if (newCategory.length) {
      setCategories((categories) => [...new Set([newCategory, ...categories])]);
    }
    setInputValue("");
  };

  const [inputValue, setInputValue] = useState("");
  return (
    <form onSubmit={handleSubmit}>
      <h2>Add Category</h2>
      <input type="text" value={inputValue} onChange={handleInput} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
