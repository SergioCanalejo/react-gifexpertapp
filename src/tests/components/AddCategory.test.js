import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Testing AddCategory component', () => {
  const setCategories = jest.fn();

  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Must show the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Must change textbox', () => {
    simulateChange();
    const input = wrapper.find('input');
    expect(input.prop('value')).toBe('Hello World!');
  });

  test('Not sumbit de form', () => {
    simulateSubmit();
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('Submit have to be called', () => {
    simulateChange();
    simulateSubmit();
    expect(setCategories).toHaveBeenCalledTimes(1);
    const input = wrapper.find('input');
    expect(input.prop('value')).toBe('');
  });

  const simulateChange = () => {
    const input = wrapper.find('input');
    const event = { target: { value: 'Hello World!' } };
    input.simulate('change', event);
  };

  const simulateSubmit = () => {
    const form = wrapper.find('form');
    const event = { preventDefault() {} };
    form.simulate('submit', event);
  };
});
