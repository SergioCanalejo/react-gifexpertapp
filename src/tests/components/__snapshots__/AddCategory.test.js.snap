// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory component Must show the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add Category
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
